libs/klv/src/main/java/org/codice/alliance/libs/klv/DistinctKlvProcessor.java

```diff
@@ -34,11 +36,15 @@ protected DistinctKlvProcessor(String attributeName, String stanagFieldName) {
 
     @Override
     protected void doProcess(Attribute attribute, Metacard metacard) {
-        metacard.setAttribute(new AttributeImpl(attributeName,
-                attribute.getValues()
-                        .stream()
-                        .distinct()
-                        .collect(Collectors.toList())));
+
+        List<Serializable> serializables = attribute.getValues()
+                .stream()
+                .distinct()
+                .collect(Collectors.toList());
+
+        if (!serializables.isEmpty()) {
+            metacard.setAttribute(new AttributeImpl(attributeName, serializables));
```

Comment:
Create a new DistinctSingleKlvProcessor that is used for attributes are defined to be single valued.